### DISCUSSION TITLE
Preload for PlayBgm Sctrl

### DISCUSSION THREAD

**Discussion by Amidweiz** [ikemen-engine/Ikemen-GO#1257] [Feature Request]
### Is your feature request related to a problem? Please describe.

As I alluded to with [my error report with bgmlength and bgmposition](https://github.com/ikemen-engine/Ikemen-GO/issues/1066) I would like to have a trigger within the PlayBgm sctrl that preloads an audio track. This would allow for more seemless music transitions than what is currently available at the moment. My laptop has a Samsung Evo 860 1TB SSD in it and the transition from one mp3 file has around a 0.5 second delay between loading each file.

### Describe the solution you'd like

`preload =` trigger with the default of 0 would suffice. Hopefully there isn't much issues with preloading tracks into memory. Maybe an sctrl to unload a track from memory when not needed could be a solution if necessary to free up memory.  

### Additional context

This was tested with the may 30th version of nightly build.

### Search terms

PlayBgm

**Comment by Lazin3ss** [comment]
We can probably preload them at compile time when looking through the SCTRLs and finding PlayBGM ones.
